### ISSUE TITLE
Different mouse cursor

### ISSUE THREAD

**Issue by Elronnd** [jwilm/alacritty#421]
At the moment, the mouse cursor just looks like normal.  Is there any chance of making it a "text field" cursor when hovering over an alacritty window?  Something like [this](https://mdn.mozillademos.org/files/3809/text.gif).

**Comment by leavengood** [comment]
This looked like an easy fix by adding `window.set_cursor(glutin::MouseCursor::Text);` to Window::new but when I tested on OS X it did not seem to have any effect.

Maybe @jwilm can provide some guidance.

**Comment by jwilm** [comment]
@leavengood sounds like a bug in glutin

**Comment by leavengood** [comment]
The cursor example works for me in master of tomaka's glutin, but your clone currently won't compile for me.

**Comment by leavengood** [comment]
I changed to the revision of your fork of glutin which is currently being used with alacritty and got the cursor example working there too. So I am not sure why that call isn't working. It may be due to when I'm calling it? I'll keep messing around.

**Comment by leavengood** [comment]
I got it working by putting the cursor change in the event handling, but I don't like how it is working so I will have to experiment more.

**Comment by DivineGod** [comment]
@leavengood I think what needs to happen is that it should check for mouse move events and update the cursor to `glutin::MouseCursor::Normal` when the cursor is outside the window and `glutin::MouseCursor::Text` when it's inside. This matches how iTerm2 is doing it.

**Comment by leavengood** [comment]
@DivineGod Yes exactly. I've found the part of the code which tracks mouse moves. We just need to track the size of the window and have a way to expose the set_cursor method there, since as far as I can see that part of the code currently doesn't have access to the window. I'm still learning the code structure so I haven't gotten that far yet.

If someone else wants to handle this, feel free. But otherwise I'll hack on it a bit over the next week.

**Comment by chrisduerr** [comment]
This seems to work fine for me on linux. For macos there's #966.
